src/coord/src/coord.rs

```diff
@@ -905,19 +945,17 @@ impl Coordinator {
                     // reduce the volume of the collection, but we don't have that
                     // information here.
                     if !index_state.upper.frontier().is_empty() {
-                        let mut compaction_frontier = Antichain::new();
-                        for time in index_state.upper.frontier().iter() {
-                            compaction_frontier.insert(
+                        // The since_handle for this GlobalId should have already been registered with
+                        // an AntichainToken. Advance it. Changes to the AntichainToken's frontier
+                        // will propagate to the Frontiers' since, and changes to that will propate to
```

Comment:
An aborted read transaction causes the frontier to stop advancing:

```
> CREATE TABLE t1 (f1 INTEGER);

> BEGIN

> SELECT * FROM t1;

> SELECT mz_internal.mz_sleep(2);
<null>

> ROLLBACK;

> SELECT mz_internal.mz_sleep(2);
<null>

! SELECT * FROM t1 AS OF NOW() - INTERVAL '1 second';
is not valid for all inputs
```

The final query should have failed becase it is trying to read as of timestamp before the expected frontier. Instead, the frontier stopped advancing at the start of the transaction and never resumed advancing after the ```ROLLBACK```.

Reply:
Good find! ROLLBACK indeed was busted.